### PR TITLE
Updating jailer example to use the jailer config

### PR DESCRIPTION
**Pull request by xibz**
Previous jailer example led to some confusion, #92, due to not knowing the need to manually set up the proper jailer workspace. New example uses the jailer config in which allows the SDK to do all the setting up.

Signed-off-by: xibz <impactbchang@gmail.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jeromegn** [comment]
@xibz are you sure this is an accurate example?

Looking at the test code, it looks like much more is involved: https://github.com/firecracker-microvm/firecracker-go-sdk/blob/5ac6d3f23ba4b362def6a064d79ad2812cb5de57/machine_test.go#L95-L249

- Copying the rootfs, vmlinux, etc. in the appropriate places.
- Setting the correct UID and GID on these files.
- Creating the log files (again, approriate UID, GID)
- Configure the jailer via JailerCommandBuilder

Your example appears to show that this is not needed, but it won't quite work without *some* of the additional steps.

You do need `EnableJailer` on the config though! Else, it won't start anything (afaict)

**Comment by xibz** [comment]
@jeromegn - Thanks for noticing the missing `EnableJailer`! I've went ahead and also included some logic to do some validation. The `NaiveChrootStrategy` handles copying the rootfs and kernel image to the correct places. The reason why the tests does this is to create a new workspace for the test. The `JailerCommandBuilder` is not needed when setting the `EnableJailer` flag and specifying the proper jailer configuration.

File: example_test.go [hunk 1]
```diff
@@ -220,51 +224,86 @@ func ExampleNetworkInterface_rateLimiting() {
...
+			ChrootBaseDir:  path,
+			ChrootStrategy: firecracker.NewNaiveChrootStrategy(pathToWorkspace, kernelImagePath),
 		},
-		DisableValidation: true,
 	}
 
-	// Passes the custom jailer command into the constructor
-	m, err := firecracker.NewMachine(ctx, cfg, firecracker.WithProcessRunner(b.Build(ctx)))
+	// This will stat the kernel image and take a look at its GID and UID. If the
+	// GID and UID do not matach the GID and UID provided in the configuration,
```
Comment (samuelkarp) [inline example_test.go]: s/matach/match/

File: example_test.go [hunk 2]
```diff
@@ -220,51 +224,86 @@ func ExampleNetworkInterface_rateLimiting() {
...
+			NumaNode:       firecracker.Int(0),
+			ChrootBaseDir:  path,
+			ChrootStrategy: firecracker.NewNaiveChrootStrategy(pathToWorkspace, kernelImagePath),
 		},
-		DisableValidation: true,
 	}
 
-	// Passes the custom jailer command into the constructor
-	m, err := firecracker.NewMachine(ctx, cfg, firecracker.WithProcessRunner(b.Build(ctx)))
+	// This will stat the kernel image and take a look at its GID and UID. If the
```
Comment (samuelkarp) [inline example_test.go]: Do we really require the UID/GID to match, or just require that the target UID/GID has permission to read the kernel image?

File: example_test.go [hunk 3]
```diff
@@ -220,51 +224,86 @@ func ExampleNetworkInterface_rateLimiting() {
...
+	}
+
+	if kernelImageInfo.Uid != uint32(uid) || kernelImageInfo.Gid != uint32(gid) {
+		panic(fmt.Errorf("Kernel image does not have the proper UID or GID.\n"+
+			"To fix this simply run:\n"+
+			"sudo chown %d:%d %s",
+			uid, gid, fcCfg.KernelImagePath))
+	}
+
+	// Check that each drive has the correct UID and GID.
```
Comment (samuelkarp) [inline example_test.go]: Same question here, though for writable drives I presume we care about the write bit as well.

File: example_test.go (hunk 2 above)
Comment (xibz) [reply example_test.go]: Yes, the jailer will return errors otherwise if the UID and GID do not match.

File: example_test.go (hunk 1 above)
Comment (xibz) [reply example_test.go]: Removed comment as it seemed unnecessary since all we need is a readable kernel image

File: example_test.go (hunk 3 above)
Comment (xibz) [reply example_test.go]: Fixed!

File: example_test.go [hunk 4]
```diff
@@ -221,52 +217,71 @@ func ExampleNetworkInterface_rateLimiting() {
...
+		panic(fmt.Errorf("Failed to open kernel image: %v", err))
+	}
+	f.Close()
+
+	// Check each drive is readable and writable
+	for _, drive := range fcCfg.Drives {
+		drivePath := firecracker.StringValue(drive.PathOnHost)
+		f, err := os.OpenFile(drivePath, os.O_RDWR, 0666)
+		if err != nil {
+			panic(fmt.Errorf("Failed to drive: %v", err))
```
Comment (sipsma) [inline example_test.go]: `Failed to open drive for reading and writing`?

Comment (xibz) [reply example_test.go]: Fixed!

**Comment by xibz** [comment]
> I'm assuming that you've run the example where you've replaced /path/to/rootfs with an actual value to verify it all works in practice? If so LGTM.

Yea, works when replacing the proper strings. I've went ahead and squashed the fix up commits

**Pull request merged**